### ISSUE TITLE
Possibly extra hwloc files in the repo

### ISSUE THREAD

**Issue by artpol84** [open-mpi/ompi#3940]
@ggouaillardet when I ran autogen on master and did `git diff` I noticed:
```shell
$ git diff
diff --git a/opal/mca/hwloc/hwloc2x/hwloc/include/private/autogen/config.h.in b/opal/mca/hwloc/hwloc2x/hwloc/include/private/autogen/config.h.in
index f65a8be..5d3ec37 100644
--- a/opal/mca/hwloc/hwloc2x/hwloc/include/private/autogen/config.h.in
+++ b/opal/mca/hwloc/hwloc2x/hwloc/include/private/autogen/config.h.in
@@ -634,9 +634,6 @@
 /* Define to 1 if scotch is netlocscotch is enabled */
 #undef NETLOC_SCOTCH
 
-/* Define to 1 if your C compiler doesn't accept -c and -o together. */
-#undef NO_MINUS_C_MINUS_O
-
 /* Name of package */
 #undef PACKAGE
``` 

it seems to me that `opal/mca/hwloc/hwloc2x/hwloc/include/private/autogen/config.h.in` shouldn't be in the repo. But it went there:
```shell
$ git log opal/mca/hwloc/hwloc2x/hwloc/include/private/autogen/config.h.in
commit 593e4ce63f22d32943de1915cd509261ccf20042
Author: Gilles Gouaillardet <gilles@rist.or.jp>
Date:   Mon Apr 10 10:12:59 2017 +0900

    hwloc: add hwloc2x
    
    internal hwloc 2x is used with --with-hwloc=future
    
    Signed-off-by: Gilles Gouaillardet <gilles@rist.or.jp>
```

Can you have a look?

**Comment by rhc54** [comment]
I have it fixed - commit coming

**Comment by rhc54** [comment]
Fixed with 7d8d877837de3987dffb9c9a3bea4224c4480f61
